contracts/babylon/src/contract.rs

```diff
@@ -188,11 +188,21 @@ fn reply_init_callback_light_client(
     reply: SubMsgResponse,
 ) -> Result<Response<BabylonMsg>, ContractError> {
     // Try to get contract address from events in reply
-    let addr = reply_init_get_contract_address(reply)?;
+    let addr = reply_init_get_contract_address(&reply)?;
+
+    // Fetch the first msg_response as the base header data of BTC light client.
+    let base_header_bytes = reply
+        .msg_responses
+        .into_iter()
+        .next()
+        .ok_or(ContractError::MissingBaseHeaderInBtcLightClientResponse)?
+        .value;
+
     CONFIG.update(deps.storage, |mut cfg| {
-        cfg.btc_light_client = Some(addr);
+        cfg.btc_light_client = Some((addr, base_header_bytes));
```

Comment:
Another approach is to query the base header from the light client contract later, not sure we have a preference over these two options.